feature/search/src/main/java/com/google/samples/apps/nowinandroid/feature/search/SearchScreen.kt

```diff
@@ -16,80 +16,252 @@
 internal fun SearchRoute(
     modifier: Modifier = Modifier,
     onBackClick: () -> Unit,
-    viewModel: SearchViewModel = hiltViewModel(),
+    onInterestsClick: () -> Unit,
+    onTopicClick: (String) -> Unit,
+    interestsViewModel: InterestsViewModel = hiltViewModel(),
+    searchViewModel: SearchViewModel = hiltViewModel(),
+    forYouViewModel: ForYouViewModel = hiltViewModel(),
 ) {
+    val uiState by searchViewModel.searchResultUiState.collectAsStateWithLifecycle()
     SearchScreen(
         modifier = modifier,
         onBackClick = onBackClick,
-        onSearchQueryChanged = viewModel::onSearchQueryChanged,
+        onFollowButtonClick = interestsViewModel::followTopic,
+        onInterestsClick = onInterestsClick,
+        onSearchQueryChanged = searchViewModel::onSearchQueryChanged,
+        onTopicClick = onTopicClick,
+        onNewsResourcesCheckedChanged = forYouViewModel::updateNewsResourceSaved,
+        uiState = uiState,
     )
 }
 
 @Composable
 internal fun SearchScreen(
     modifier: Modifier = Modifier,
     onBackClick: () -> Unit = {},
+    onFollowButtonClick: (String, Boolean) -> Unit = { _, _ -> },
+    onInterestsClick: () -> Unit = {},
+    onNewsResourcesCheckedChanged: (String, Boolean) -> Unit = { _, _ -> },
     onSearchQueryChanged: (String) -> Unit = {},
+    onTopicClick: (String) -> Unit = {},
+    uiState: SearchResultUiState = SearchResultUiState.Loading,
 ) {
+    val searchQuery = remember { mutableStateOf("") }
     TrackScreenViewEvent(screenName = "Search")
-    Column(
-        modifier = modifier,
-        horizontalAlignment = Alignment.CenterHorizontally,
-    ) {
+    Column(modifier = modifier) {
         Spacer(Modifier.windowInsetsTopHeight(WindowInsets.safeDrawing))
         SearchToolbar(
             onBackClick = onBackClick,
             onSearchQueryChanged = onSearchQueryChanged,
+            searchQuery = searchQuery,
         )
+        when (uiState) {
+            SearchResultUiState.Loading -> Unit
+            SearchResultUiState.EmptyQuery -> Unit
+            is SearchResultUiState.Success -> {
+                if (uiState.isEmpty()) {
+                    EmptySearchResultBody(
+                        onInterestsClick = onInterestsClick,
+                        searchQuery = searchQuery,
+                    )
+                } else {
+                    SearchResultBody(
+                        topics = uiState.topics,
+                        onFollowButtonClick = onFollowButtonClick,
+                        onNewsResourcesCheckedChanged = onNewsResourcesCheckedChanged,
+                        onTopicClick = onTopicClick,
+                        newsResources = uiState.newsResources,
+                    )
+                }
+            }
+        }
         Spacer(Modifier.windowInsetsBottomHeight(WindowInsets.safeDrawing))
     }
 }
 
+@Composable
+fun EmptySearchResultBody(
+    onInterestsClick: () -> Unit = {},
+    searchQuery: MutableState<String>,
```

Comment:
Do not pass MutableState to Composable function.
```suggestion
    searchQuery: String,
```

Reply:
Done